timezone/init.sls

```diff
@@ -1,7 +1,7 @@
 # This state configures the timezone.
 
-{%- set timezone = salt['pillar.get']('timezone:name', 'Europe/Berlin') %}
-{%- set utc = salt['pillar.get']('timezone:utc', True) %}
+{%- set timezone = salt['pillar.get']('timezone:lookup:name', 'Europe/Berlin') %}
+{%- set utc = salt['pillar.get']('timezone:lookup:utc', True) %}
 {% from "timezone/map.jinja" import confmap with context %}
```

Comment:
Even considering the current state of `map.jinja` file, combining the "lookup" and top-level values is still easy.
Like:
```jinja
{%- set timezone = salt['pillar.get']('timezone:name', confmap.name) %}
{%- set utc = salt['pillar.get']('timezone:utc', confmap.utc) %}
```
Putting it after the `from` statement, we can achieve flexibility and compatibility, plus getting rid of hard-coded defaults and use the ones from `defaults.yaml` file. This might satisfy your use case.